packages/flame/lib/src/components/text_box_component.dart

```diff
@@ -79,6 +79,8 @@ class TextBoxComponent<T extends TextRenderer> extends TextComponent {
 
   @visibleForTesting
   Image? cache;
+  Function(double)? newLineCallback;
```

Comment:
This one should probably be marked as `@internal`? Or we could make it a `ValueNotifier` so that users also can listen to it, what do you think?

Reply:
I thought the same, but then I became unsure: Are there scenarios with a `TextBoxComponent` where reacting to line changes is needed? Also, wouldn't it be beneficial to transfer the `onComplete` function to `TextBoxComponent` so that users can benefit from it?

Reply:
Moving `onComplete` to `TextBoxComponent` sounds like it would make a lot of sense!

> Are there scenarios with a TextBoxComponent where reacting to line changes is needed?

I think there is, especially if you have a very text-driven game.

Reply:
Okay, I'll do it later. I would also switch `newLineCallback` to a `ValueNotifier`.

Reply:
Super!